docs/components/plugins/AppConfigPropsTable.js

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { css } from 'react-emotion';
+import { InlineCode } from '~/components/base/code';
+import MDX from '@mdx-js/runtime';
+import * as components from '~/common/translate-markdown';
+
+const STYLES_TABLE = css`
+  font-size: 1rem;
+  margin-top: 24px;
+`;
+
+const STYLES_DESCRIPTION_CELL = css`
+  word-break: break-word;
+  white-space: break-spaces;
+`;
+
+//appends a property and recursivley calls itself to append sub-properties, accounting for nested "level"
+function appendProp(formattedSchema, property, _level) {
+  let level = _level;
+  const propertyKey = property[0];
+  const propertyValue = property[1];
+
+  //don't append deprecated or "hidden" properties
+  if (propertyValue.meta && (propertyValue.meta.deprecated || propertyValue.meta.hidden)) {
+    return;
+  }
+
+  //append passed-in property
+  formattedSchema.push({
+    //The ` quotes are for markdown highlighting, and #### is to apply anchor linking only on top-level properties
+    name: level ? `\`${propertyKey}\`` : `#### \`${propertyKey}\``,
+    type: propertyValue.enum ? 'enum' : propertyValue.type,
+    description: smartlyCreateDescription(property),
+    level: level,
+  });
+
+  //increase nesting level for sub-properties
+  level++;
+
+  //recursively apply appending logic for each sub-property (if any) -> Note: sub-props are sometimes nested within "items"
+  if (propertyValue.properties) {
+    Object.entries(propertyValue.properties).forEach(subproperty => {
+      appendProp(formattedSchema, subproperty, level);
+    });
+  } else if (propertyValue.items && propertyValue.items.properties) {
+    Object.entries(propertyValue.items.properties).forEach(subproperty => {
+      appendProp(formattedSchema, subproperty, level);
+    });
+  }
+}
+
+//setting up a property's formatted description value, with all the possible extra values
+function smartlyCreateDescription(property) {
```

Comment:
```suggestion
function createDescription(property) {
```

`smartly` here does not really help me understand what this function does.